esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp

```diff
@@ -157,10 +157,48 @@ bool isESDLDefinitionBound(const char * esdldefname, int version)
     return isESDLDefinitionBound(id);
 }
 
+
+void fetchESDLDefinitionFromDaliByNameOnly(const char * name, StringBuffer & def)
+{
+    if (!name || !*name)
+        throw MakeStringException(-1, "Unable to fetch ESDL Service definition information, definition name is not available");
+
+    DBGLOG("ESDL Binding: Fetching ESDL Definition from Dali based on name: %s ", name);
+
+    Owned<IRemoteConnection> conn = querySDS().connect(ESDL_DEFS_ROOT_PATH, myProcessSession(), RTM_LOCK_READ, SDS_LOCK_TIMEOUT_DESDL);
+    if (!conn)
+       throw MakeStringException(-1, "Unable to connect to ESDL Service definition information in dali '%s'", ESDL_DEFS_ROOT_PATH);
+
+    conn->close(false);//release lock right away
```

Comment:
This seems a little strange.  It looks like EsdlConfig is the only service that does it quite this way.. is it kosher?  @jakesmith can you comment?